From 2042024e57a804a580f912b8487270c3bcc1d3f1 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 18 Dec 2012 11:51:42 +0000
Subject: [PATCH 002/102] daemon: Make gdisk into an optional dependency and
 optgroup.

Also document that gdisk is a dependency at all.

This fixes commit 956e30effa25eee00fb9e8f5303a6a1fe7409037.
(cherry picked from commit fa162417ed97218d528b8fa03cd22ad9ce788a56)
---
 README               | 3 +++
 daemon/parted.c      | 7 +++++++
 generator/actions.ml | 2 ++
 3 files changed, 12 insertions(+)

diff --git a/README b/README
index de2059e..7a19413 100644
--- a/README
+++ b/README
@@ -118,6 +118,9 @@ For basic functionality and the C tools:
 - yajl >= 2 (optional)
   JSON parser, needed to handle the output of ldmtool.
 
+- gdisk (optional)
+  For managing some aspects of GPT disks.
+
 - netpbm, icoutils (optional)
   These programs are used to render icons from guests.
 
diff --git a/daemon/parted.c b/daemon/parted.c
index 5a6d84f..b1ece1b 100644
--- a/daemon/parted.c
+++ b/daemon/parted.c
@@ -27,6 +27,7 @@
 
 #include "daemon.h"
 #include "actions.h"
+#include "optgroups.h"
 
 GUESTFSD_EXT_CMD(str_parted, parted);
 GUESTFSD_EXT_CMD(str_sfdisk, sfdisk);
@@ -752,6 +753,12 @@ do_part_set_mbr_id (const char *device, int partnum, int idbyte)
 }
 
 int
+optgroup_gdisk_available (void)
+{
+  return prog_exists (str_sgdisk);
+}
+
+int
 do_part_set_gpt_type(const char *device, int partnum, const char *guid)
 {
   if (partnum <= 0) {
diff --git a/generator/actions.ml b/generator/actions.ml
index e479f03..1530040 100644
--- a/generator/actions.ml
+++ b/generator/actions.ml
@@ -10732,6 +10732,7 @@ you are better to use C<guestfs_mv> instead." };
     name = "part_set_gpt_type";
     style = RErr, [Device "device"; Int "partnum"; String "guid"], [];
     proc_nr = Some 392;
+    optional = Some "gdisk";
     tests = [];
     shortdesc = "set the type GUID of a GPT partition";
     longdesc = "\
@@ -10746,6 +10747,7 @@ for a useful list of type GUIDs." };
     name = "part_get_gpt_type";
     style = RString "guid", [Device "device"; Int "partnum"], [];
     proc_nr = Some 393;
+    optional = Some "gdisk";
     tests = [
       InitGPT, Always, TestOutput (
         [["part_set_gpt_type"; "/dev/sda"; "1";
-- 
1.9.3

